go.mod

```diff
@@ -1,7 +1,9 @@
-module github.com/argoproj/argo
+module github.com/argoproj/argo/v2
 
 go 1.13
 
+replace github.com/argoproj/argo/v2 => ./
+
```

Comment:
This is a key change that simplifies the transition

Reply:
This is not a valid use of the replace directive. The result is that your packages are unusable outside of this module.

Reply:
You _have_ to update all of the argo import paths in all source files and add a v2 component to them.

Reply:
Related: https://github.com/golang/go/issues/34417

Reply:
> You have to update all of the argo import paths in all source files and add a v2 component to them.

@peterbourgon I must be misunderstanding you, because we do update all the argo import paths and add a `v2/` to them

Can you reach out to me on the ArgoProj slack so we can chat about this please? https://argoproj.github.io/community/join-slack I'm `@simon`. I want to resolve this ASAP

Reply:
I followed the pattern used by `etcd`, although their case seems to be a "repository of modules" instead of a single module, which could explain the difference. If the hesitation is the module referencing itself, then we could simply move all the code to a `v2` folder as an easy change. All of the import paths already have `v2` in them

Reply:
Oops! I somehow got a view of the PR/repo where the import statements weren't updated. But I see now that they are. Mea culpa for that bit. The issue with the replace directive is being addressed in #4959 👍